core/ledger/kvledger/pvtrwstorage/transient_store.go

```diff
@@ -16,26 +16,154 @@ limitations under the License.
 
 package pvtrwstorage
 
-// TransientStoreProvider provides handle to specific 'TransientStore' that in turn manages
-// private read-write sets for a namespace
+import (
+	commonledger "github.com/hyperledger/fabric/common/ledger"
+	"github.com/hyperledger/fabric/common/ledger/util/leveldbhelper"
+	"github.com/hyperledger/fabric/core/ledger"
+	"github.com/hyperledger/fabric/core/ledger/ledgerconfig"
+	"github.com/syndtr/goleveldb/leveldb/iterator"
+)
+
+const (
+	PRWSET_NS     string = "P" // namespace for storing private read-write set in transient store.
+	PURGEINDEX_NS string = "I" // namespace for storing index on private read-write set using endorsement block height.
+)
+
+var emptyValue = []byte{}
+
+// TransientStoreProvider encapsulates a leveldb provider which is used to store
+// private read-write set of simulated transactions, and implements TransientStoreProvider
+// interface.
+type transientStoreProvider struct {
+	dbProvider *leveldbhelper.Provider
+}
+
+// TransientStoreProvider provides an instance of a TransientStore
 type TransientStoreProvider interface {
```

Comment:
Just thinking again do we need multiple logical dbs for transient store. Or one is good enough. If we use only one logical db, the ledgerid needs to be explicitly added to the composite key. We can leave this as is for now... Lets discuss the pro/cons and can change later in a separate PR